drivers/dma/dma_mcux_lpc.c

```diff
@@ -527,7 +560,17 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 			local_block.next_block = block_config->next_block;
 			local_block.source_reload_en = reload;
 
-			if (dma_mcux_lpc_queue_descriptors(data, &local_block, src_inc, dst_inc)) {
+			if (block_config->next_block == NULL) {
+				/* This is the last block. If complete callback is called only
+				 * on the last block then enable it for the last block
+				 */
+				if (!complete_callback) {
+					complete_callback = true;
+				}
```

Comment:
maybe the `if(!complete_callback)` is not needed here since in this case we are always setting it to true it seems like

Reply:
We still need this piece of code as the earlier set to true was for the case when we queue only 1 block.